### PR TITLE
Delete tag retention rule and tag immutable rule when deleting project

### LINKED ISSUE #18250: The tag retention and tag immutable policies are not deleted after deleting the project
**Steps to reproduce the problem:**
1. Create a project
2. Create tag retention and tag immutable policies to project
3. Set tag retention policies policy schedule
4. Delete this project
5. tag retention and tag immutable policies still exist
图片

**Versions:**
Please specify the versions of following systems.

- harbor version: v2.5.x ~ v2.7.x

### PR DESCRIPTION
**Pull request by stonezdj**
fixes #18250

Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
Fixes #(issue)

Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [x] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [x] Made sure tests are passing and test coverage is added if needed.
- [x] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

**Comment by chlins** [comment]
The `src/controller/event/handler/internal/artifact.go` is the internal event controller for the artifact, should only process the events related to artifact, so for project I think we should add a new file `project.go` to handle, and will be more extensible for other project events in the future.

File: src/pkg/retention/dao/retention.go [hunk 1]
```diff
@@ -67,3 +68,16 @@ func GetPolicy(ctx context.Context, id int64) (*models.RetentionPolicy, error) {
 	}
 	return p, nil
 }
+
+// ListPolicy list retention policy by query
+func ListPolicy(ctx context.Context, query *q.Query) ([]*models.RetentionPolicy, error) {
```
Comment (chlins) [inline src/pkg/retention/dao/retention.go:73]: `ListPolicies`?

**Review by chlins — APPROVED** [review]
lgtm

File: src/pkg/retention/dao/retention.go (hunk 1 above)
Comment (stonezdj) [reply src/pkg/retention/dao/retention.go:73]: Updated

**Review by wy65701436 — APPROVED** [review]
lgtm

**Pull request merged**